AutomatedTesting/Gem/PythonTests/Terrain/EditorScripts/TerrainSystem_VegetationSpawnsOnTerrainSurfaces.py

```diff
@@ -0,0 +1,217 @@
+    import azlmbr.bus as bus
+    import azlmbr.math as math
+
+    import azlmbr.areasystem as areasystem
+    import azlmbr.editor as editor
+    import azlmbr.vegetation as vegetation
+    import azlmbr.terrain as terrain
+    import azlmbr.entity as EntityId
+    import azlmbr.surface_data as surface_data
+
+    import editor_python_test_tools.hydra_editor_utils as hydra
+    from editor_python_test_tools.utils import Report
+    from editor_python_test_tools.utils import TestHelper as helper
+
+    def create_entity_at(entity_name, components_to_add, x, y, z):
+        entity = hydra.Entity(entity_name)
+        entity.create_entity(math.Vector3(x, y, z), components_to_add)
+
+        return entity
+
+    def FindHighestAndLowestZValuesInArea(aabb):
+        vegetation_items = areasystem.AreaSystemRequestBus(bus.Broadcast, 'GetInstancesInAabb', aabb)
+
+        lowest_z = sys.float_info.max
+        highest_z = sys.float_info.min
+
+        for item in vegetation_items:
+            if item.position.z < lowest_z:
+                lowest_z = item.position.z
+            if item.position.z > highest_z:
+                highest_z = item.position.z
+
+        return highest_z, lowest_z
+
+    helper.init_idle()
+
+    # Open an empty level.
+    helper.open_level("Physics", "Base")
+    helper.wait_for_condition(lambda: general.get_current_level_name() == "Base", 2.0)
+    
+    general.idle_wait_frames(1)
+
+    box_height = 20.0
+    box_dimensions = math.Vector3(20.0, 20.0, box_height)
+
+    # Add Terrain Rendering
+    hydra.add_level_component("Terrain World")
+    hydra.add_level_component("Terrain World Renderer")
+
+    # Create two terrain entities at adjoining positions
+    terrain_entity_1 = create_entity_at("Terrain1", ["Terrain Layer Spawner", "Axis Aligned Box Shape", "Terrain Height Gradient List", "Terrain Surface Gradient List"], 0.0, 0.0, box_height/2.0)
+    terrain_entity_1.get_set_test(1, "Axis Aligned Box Shape|Box Configuration|Dimensions", box_dimensions)
+
+    terrain_entity_2 = create_entity_at("Terrain2", ["Terrain Layer Spawner", "Axis Aligned Box Shape", "Terrain Height Gradient List", "Terrain Surface Gradient List"], 20.0, 0.0, box_height/2.0)
+    terrain_entity_2.get_set_test(1, "Axis Aligned Box Shape|Box Configuration|Dimensions", box_dimensions)
+
+    # Create two gradient entities.
+    gradient_value_1 = 0.25
+    gradient_value_2 = 0.5
+
+    gradient_entity_1 = create_entity_at("Gradient1", ["Constant Gradient"], 0.0, 0.0, 0.0)
+    gradient_entity_1.get_set_test(0, "Configuration|Value", gradient_value_1)
+
+    gradient_entity_2 = create_entity_at("Gradient2", ["Constant Gradient"], 0.0, 0.0, 0.0)
+    gradient_entity_2.get_set_test(0, "Configuration|Value", gradient_value_2)
+
+    # Assign mappings to the gradient surface lists, but don't assign SurfaceTags yet.
+    surface_tags: dict = {"test_tag2": 1797181423, "test_tag3": 471465849, "terrain": 3363197873}
```

Comment:
A comment to explain where these numbers come from would be helpful. Are they just arbitrary?

Reply:
Actually, looking further, I'd just remove this line entirely, especially since the terrain tag isn't even being used.  Replacement suggestions below.

Reply:
The SetComponentProperty bus call requires a u32 to be passed in. I've exposed the u32() method of SurfaceTag so I can use that.

Reply:
Can you elaborate any further?  The following tests seem to work without needing the u32() exposed:
AutomatedTesting\Gem\PythonTests\largeworlds\dyn_veg\EditorScripts\SurfaceMaskFilterOverrides_MultipleDescriptorOverridesPlantAsExpected.py
AutomatedTesting\Gem\PythonTests\largeworlds\dyn_veg\EditorScripts\PhysXColliderSurfaceTagEmitter_E2E_Editor.py

Are these tests doing something different?

Reply:
You're quite right, I can go back a step in the path and pass in the SurfaceTag, I was looking at the wrong examples...

Reply:
You're quite right, walking back a step in the path allows me to use the SurfaceTag directly. I was looking at largeworlds/dyn_veg/SurfaceMaskFilter_ExclusionList.py for examples, which does it by passing in CRCs.

Reply:
Heh, we should probably get that one cleaned up sometime too.